cabby/model/text/dual_encoder/dataset_run.py

```diff
@@ -0,0 +1,180 @@
+# Copyright 2020 The Flax Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+sys.path.append("/home/tzuf_google_com/dev/cabby")
+
+from typing import Dict
+
+from absl import logging
+import numpy as np
+import os
+import pandas as pd
+
+import torch
+from transformers import DistilBertTokenizerFast
+
+from cabby.geo import util as gutil
+from cabby.model.text import util 
+
+from cabby.geo import regions
+from cabby.model.text.dual_encoder import dataset_item
+
+tokenizer = DistilBertTokenizerFast.from_pretrained('distilbert-base-uncased')
+
+CELLID_DIM = 64
+
+
+class TextGeoSplit(torch.utils.data.Dataset):
+  """A split of of the RUN dataset.
+  
+  `points`: The ground true end-points of the samples.
+  `labels`: The ground true label of the cellid.
+  `cellids`: The ground truth S2Cell id.
+  `neighbor_cells`: One neighbor cell id of the ground truth S2Cell id.
+  `far_cells`: One far away cell id (in the region defined) of the ground truth S2Cell id.
+  """
+  def __init__(self, data: pd.DataFrame, s2level: int, 
```

Comment:
IIRC, there is a lot of duplicated code from other datasets here. If there is a fast way to reduce duplication, that'd be great. That said, it's okay to move fast and not worry much about that for now.